api-server/models.py

```diff
@@ -17,15 +24,133 @@ class State(str, Enum):
 
 
 class Player(models.Model):
+    MAX_REPOS_TO_GENERATE_FILES = 5
+    SUPPORTED_EXTENSIONS = [
+        "py",
+        "js",
+        "ts",
+        "jsx",
+        "tsx",
+        "go",
+        "dart",
+        "java",
+        "cc",
+        "cpp",
+        "c",
+        "swift",
+    ]
+
     class ConnectionState(str, Enum):
         CONNECTED = "connected"
         DISCONNECTED = "disconnected"
 
+    # id will be of the format "{session_id}-{username}"
     id = fields.CharField(max_length=40, pk=True)
-    username = fields.CharField(max_length=30, null=False, required=True)
-    connection_state = fields.CharEnumField(
-        ConnectionState, null=False, required=True, max_length=20
-    )
+    username = fields.CharField(max_length=30)
+    connection_state = fields.CharEnumField(ConnectionState, max_length=20)
     session: fields.ForeignKeyRelation[Session] = fields.ForeignKeyField(
         "models.Session", "players"
     )
+
+    # todo(ramko9999): if necessary, add the next_page number to fetch the next set of repos as an db field
+    repos: fields.ReverseRelation["Repository"]
```

Comment:
what's a reverse relation?

Reply:
Is that just saying that a `Repository` has a FK on `Player`

Reply:
Also noticed, that you are using this below. Doing this allows us to access all the repos from this Player?

Reply:
I think `ReverseRelation` allows for a simple way to get the table rows that have a foreign key pointing to the rows of the table in question. Since `Repository` has a foreign key on `Player`, I believe through the `ReverseRelation`, for a `Player` P, we can do `P.repos` to get the query set of the repositories who have foreign key pointing to P.